### PR TITLE
KAFKA-7849: Fix the warning when using GlobalKTable  

### PR DESCRIPTION
**Pull request by omarsmak**
*More detailed description of your change,
This PR intend to fix the issue in KAFKA-7849 by filtering out the global topics from the main consumer, thus the main consumer won't subscribe to any of the global topics

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java [hunk 1]
```diff
@@ -1194,7 +1195,8 @@ synchronized Pattern sourceTopicPattern() {
             final List<String> allSourceTopics = new ArrayList<>();
             if (!nodeToSourceTopics.isEmpty()) {
                 for (final List<String> topics : nodeToSourceTopics.values()) {
-                    allSourceTopics.addAll(maybeDecorateInternalSourceTopics(topics));
+                    final List<String> filteredTopics = filterAndRemoveGlobalSourceTopics(topics);
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java]: Why not just calling `allSourceTopics.removeAll(globalTopics);` ? 

Also, we should do this only once after the for-loop.

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java [hunk 2]
```diff
@@ -243,6 +243,19 @@ public void testSourceTopics() {
         assertEquals(expectedPattern.pattern(), builder.sourceTopicPattern().pattern());
     }
 
+    @Test
+    public void testSourceTopicsWithGlobalTopics() {
+        builder.setApplicationId("X");
+        builder.addSource(null, "source-1", null, null, null, "topic-1");
+        builder.addSource(null, "source-2", null, null, null, "topic-2");
+        builder.addGlobalStore(new MockKeyValueStoreBuilder("global-store", false).withLoggingDisabled(), "globalSource", null, null, null, "globalTopic", "global-processor", new MockProcessorSupplier());
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java]: nit: line too long
```
builder.addGlobalStore(
    new MockKeyValueStoreBuilder("global-store", false).withLoggingDisabled(),
    "globalSource",
    null,
    null,
    null,
    "globalTopic",
    "global-processor",
    new MockProcessorSupplier());
```

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java [hunk 3]
```diff
@@ -243,6 +243,19 @@ public void testSourceTopics() {
...
+
+
+        final Pattern expectedPattern = Pattern.compile("topic-1|topic-2");
+
+        assertEquals(expectedPattern.pattern(), builder.sourceTopicPattern().pattern());
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java]: nit: use `assertThat(builder.sourceTopicPattern().pattern(), equalTo(expectedPattern));

**Review by mjsax — COMMENTED** [review]
Thanks for the PR @omarsmak! Just some nits.

**Comment by mjsax** [comment]
Test failures unrelated and tracked in Jira.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java (hunk 1 above)
Comment (omarsmak) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java]: True, fixed it

**Comment by omarsmak** [comment]
Thanks a lot @mjsax ! I have committed the code with the advised modifications .

**Comment by mjsax** [comment]
All three builds failed (results are not available any longer).

Retest this please.

Call for second review @guozhangwang @bbejeck @vvcephei @ableegoldman @cadonna @abbccdda

**Comment by omarsmak** [comment]
Hey @mjsax, I have restarted all tests, I am not really if this is related to the patch, can you please take a look? Thanks

**Review by mjsax — APPROVED** [review]
LGTM.

Call for second review @guozhangwang @bbejeck @vvcephei @ableegoldman @cadonna @abbccdda

**Comment by mjsax** [comment]
Java 11 / 2.12 passed -- other two runs failed (test results are not available any longer).

Retest this please.

**Review by guozhangwang — APPROVED** [review]
LGTM!

**Comment by omarsmak** [comment]
Hey @mjsax , I have re-triggered all the tests, I hope this time will pass !

**Comment by omarsmak** [comment]
Finally all tests have been passed!

**Comment by omarsmak** [comment]
Hey @mjsax , can we have this one merged soon? Thanks

**Comment by omarsmak** [comment]
Hey @mjsax @guozhangwang, sorry for commenting again, I was just wondering what is the next plan for this PR?

**Comment by mjsax** [comment]
Thanks for the PR @omarsmak! Merged to `trunk`.

**Comment by omarsmak** [comment]
Thank you @mjsax

**Pull request merged**